app/src/main/java/org/mozilla/fenix/FenixApplication.kt

```diff
@@ -162,10 +163,31 @@ open class FenixApplication : LocaleAwareApplication() {
         //    runStorageMaintenance()
         // }
 
+        val taskQueue = components.performance.visualCompletenessQueue
         registerActivityLifecycleCallbacks(
-            PerformanceActivityLifecycleCallbacks(components.performance.visualCompletenessQueue)
+            PerformanceActivityLifecycleCallbacks(taskQueue)
         )
 
+        // Enable the service-experiments component to be initialized after visual completeness
+        // for performance wins.
+        if (settings().isExperimentationEnabled && Config.channel.isReleaseOrBeta) {
+            taskQueue.runIfReadyOrQueue {
+                Experiments.initialize(
+                    applicationContext = applicationContext,
+                    configuration = mozilla.components.service.experiments.Configuration(
+                        httpClient = components.core.client,
+                        kintoEndpoint = KINTO_ENDPOINT_PROD
+                    )
+                )
+            }
+        } else {
+            // We should make a better way to opt out for when we have more experiments
+            // See https://github.com/mozilla-mobile/fenix/issues/6278
+            ExperimentsManager.optOutSearchWidgetExperiment(this)
+        }
+
+        ExperimentsManager.initSearchWidgetExperiment(this)
```

Comment:
I don't think this will work correctly – I assume the experiments system needs to be initialized before this can be called. Furthermore, it probably has some impact on startup (given we're reading sharedPrefs on the main thread). I think we can resolve both of these issues if this is run after `Experiments.initialize` in the visualCompleteness queue.